image/image.go

```diff
@@ -125,25 +126,51 @@ func Prepare(opts *Options) error {
 	return bootstrapToRootDir(sto, model, opts, local)
 }
 
-// these are postponed, not implemented or abandoned, not finalized,
-// don't let them sneak in into a used model assertion
-var reserved = []string{"core", "os", "class", "allowed-modes"}
+func modelAssertion(opts *Options) (*asserts.Model, error) {
+	switch {
+	case opts.ModelFile != "" && opts.ModelRef != "":
+		return nil, fmt.Errorf("specifying both model file and model reference is not allowed")
+	case opts.ModelFile != "":
+		return modelAssertionFromFile(opts.ModelFile)
+	case opts.ModelRef != "":
+		return modelAssertionFromStore(opts.ModelRef)
+	}
 
-func decodeModelAssertion(opts *Options) (*asserts.Model, error) {
-	fn := opts.ModelFile
+	return nil, fmt.Errorf("need either a model file or a model reference")
+}
 
-	rawAssert, err := ioutil.ReadFile(fn)
+func modelAssertionFromFile(fn string) (*asserts.Model, error) {
+	raw, err := ioutil.ReadFile(fn)
 	if err != nil {
-		return nil, fmt.Errorf("cannot read model assertion: %s", err)
+		return nil, fmt.Errorf("cannot read model assertion %q: %s", fn, err)
 	}
-
-	ass, err := asserts.Decode(rawAssert)
+	rawAssert, err := asserts.Decode(raw)
 	if err != nil {
 		return nil, fmt.Errorf("cannot decode model assertion %q: %s", fn, err)
 	}
+
+	return decodeModelAssertion(rawAssert)
+}
+
+func modelAssertionFromStore(modelRef string) (*asserts.Model, error) {
+	assSto := store.New(nil, nil)
+
+	l := strings.Split(modelRef, "/")
```

Comment:
We don't use such a foo/bar/baz syntax in any other stable API. With this API, people need to remember what the exact parameter needed are, and their order, and then provide in a non-conventional way. These would be better as proper parameters (--brand-id, etc).